doc/source/developing/testing.rst

```diff
@@ -3,510 +3,415 @@
+This pytest invocation will only run the tests defined by the
+``TestSetWidth`` class. See the `pytest documentation <https://docs.pytest.org/en/6.2.x/usage.html>`_ for more on the various ways to invoke pytest.
 
 Finally, to determine which test is failing while the tests are running, it helps
 to run the tests in "verbose" mode. This can be done by passing the ``-v`` option
-to the ``nosetests`` executable.
+to the ``pytest`` executable.
+
+.. code-block:: bash
+
+    $ pytest -v
 
-All of the above ``nosetests`` options can be combined. So, for example to run
+All of the above ``pytest`` options can be combined. So, for example, to run
 the ``TestSetWidth`` tests with verbose output, letting the output of print
 statements come out on the terminal prompt, and enabling pdb debugging on errors
 or test failures, one would do:
 
 .. code-block:: bash
 
-    $ nosetests --pdb --pdb-failures -v -s yt.visualization.tests.test_plotwindow:TestSetWidth
+    $ pytest yt/visualization/tests/test_plotwindow.py::TestSetWidth -v -s --pdb
+
+More pytest options can be found by using the ``--help`` flag
+
+.. code-block:: bash
+
+    $ pytest --help
 
 .. _answer_testing:
 
 Answer Testing
 --------------
 
-What do Answer Tests Do
+What Do Answer Tests Do
 ^^^^^^^^^^^^^^^^^^^^^^^
 
-Answer tests test **actual data**, and many operations on that data, to make
-sure that answers don't drift over time.  This is how we test frontends, as
-opposed to operations, in yt.
+Answer tests use `actual data <https://yt-project.org/data/>`_ to test reading, writing, and various manipulations of that data. Answer tests are how we test frontends, as opposed to operations, in yt.
+
+In order to ensure that each of these operations are performed correctly, we store gold standard versions of yaml files called answer files. More generally, an answer file is a yaml file containing the results of having run the answer tests, which can be compared to a reference, enabling us to control that results do not drift over time.
 
 .. _run_answer_testing:
 
 How to Run the Answer Tests
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The very first step is to make a directory and copy over the data against which
-you want to test. Next, add the config parameter ``test_data_dir`` pointing to
-directory with the test data you want to test with, e.g.:
+In order to run the answer tests locally:
+
+* Create a directory to hold the data you'll be using for the answer tests you'll be writing or the answer tests you'll be running. This directory should be outside the yt git repository in a place that is logical to where you would normally store data.
+
+* Add folders of the required data to this directory. Other yt data, such as ``IsolatedGalaxy``, can be downloaded to this directory as well.
+
+* Tell yt where it can find the data. This is done by setting the config parameter ``test_data_dir`` to the path of the
+directory with the test data downloaded from https://yt-project.org/data/. For example,
```

Comment:
I think this line is causing the one and only new warning in the docs build:
```
/var/jenkins_home/workspace/ya_docs_new/doc/source/developing/testing.rst:174: WARNING: Bullet list ends without a blank line; unexpected unindent.
```
here's a suggestion to fix it
```suggestion
directory with the test data downloaded from `the yt hub <https://yt-project.org/data/>`_. For example,
```